File: Knjigoteka.Model/Entities/Reservation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Knjigoteka.Model.Entities
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int BranchId { get; set; }
        [Required]
        public DateTime ReservedAt { get; set; }
        public DateTime? ClaimedAt { get; set; }
        public DateTime? ReturnedAt { get; set; }
        public DateTime? ExpiredAt { get; set; }
        [Required]
        public ReservationStatus Status { get; set; } = ReservationStatus.Pending;

        [ForeignKey(nameof(BookId))]
        public Book Book { get; set; } = null!;
        [ForeignKey(nameof(UserId))]
        public User User { get; set; } = null!;
        [ForeignKey(nameof(BranchId))]
        public Branch Branch { get; set; } = null!;
    }
    public enum ReservationStatus
    {
        Pending,
        Claimed,
        Returned,
        Expired
    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/reservation.dart
class Reservation {
  final int id;
  final int userId;
  final String userName;
  final int bookId;
  final String bookTitle;
  final int branchId;
  final String branchName;
  final DateTime reservedAt;
  final DateTime? claimedAt;
  final DateTime? returnedAt;
  final DateTime? expiredAt;
  final String status;

  Reservation({
    required this.id,
    required this.userId,
    required this.userName,
    required this.bookId,
    required this.bookTitle,
    required this.branchId,
    required this.branchName,
    required this.reservedAt,
    this.claimedAt,
    this.returnedAt,
    this.expiredAt,
    required this.status,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      userId: json['userId'],
      userName: json['userName'] ?? '',
      bookId: json['bookId'],
      bookTitle: json['bookTitle'] ?? '',
      branchId: json['branchId'],
      branchName: json['branchName'] ?? '',
      reservedAt: DateTime.parse(json['reservedAt']),
      claimedAt: json['claimedAt'] != null
          ? DateTime.parse(json['claimedAt'])
          : null,
      returnedAt: json['returnedAt'] != null
          ? DateTime.parse(json['returnedAt'])
          : null,
      expiredAt: json['expiredAt'] != null
          ? DateTime.parse(json['expiredAt'])
          : null,
      status: json['status'] ?? '',
    );
  }
}
